public/app/plugins/datasource/cloud-monitoring/module.ts

```diff
@@ -1,14 +1,65 @@
-import { DataSourcePlugin } from '@grafana/data';
+import { DataSourcePlugin, DashboardLoadedEvent } from '@grafana/data';
+import { getAppEvents } from '@grafana/runtime';
 
 import CloudMonitoringCheatSheet from './components/CloudMonitoringCheatSheet';
 import { ConfigEditor } from './components/ConfigEditor/ConfigEditor';
 import { QueryEditor } from './components/QueryEditor';
 import { CloudMonitoringVariableQueryEditor } from './components/VariableQueryEditor';
 import CloudMonitoringDatasource from './datasource';
-import { CloudMonitoringQuery } from './types';
+import pluginJson from './plugin.json';
+import { trackCloudMonitoringDashboardLoaded } from './tracking';
+import { CloudMonitoringQuery, QueryType } from './types';
 
 export const plugin = new DataSourcePlugin<CloudMonitoringDatasource, CloudMonitoringQuery>(CloudMonitoringDatasource)
   .setQueryEditorHelp(CloudMonitoringCheatSheet)
   .setQueryEditor(QueryEditor)
   .setConfigEditor(ConfigEditor)
   .setVariableQueryEditor(CloudMonitoringVariableQueryEditor);
+
+// Track dashboard loads to RudderStack
+getAppEvents().subscribe<DashboardLoadedEvent<CloudMonitoringQuery>>(
+  DashboardLoadedEvent,
+  ({ payload: { dashboardId, orgId, grafanaVersion, queries } }) => {
+    const cloudmonitorQueries = queries[pluginJson.id];
+    let stats = {
+      [QueryType.TIME_SERIES_QUERY]: {
+        hidden: 0,
```

Comment:
I don't think we need to check hidden queries since they are not executed. There is no useful info we could get from it.

Reply:
Yep that's a good point.